### DISCUSSION TITLE
How to move Alpine.store into a module

### DISCUSSION THREAD

**Discussion by Ravenna** [alpinejs/alpine#3148] [1. Help]
I have a pretty straight forward Alpine Store. And it works well.  I am trying to move everything to be modularized so that I can distribute it easier.

My current store: 
`
document.addEventListener('alpine:init', () => {
    Alpine.store('foo', {
        foo: Alpine.$persist(''),
        bar: Alpine.$persist(false),

        bat(address){
            this.foo = address;
        },        
    });

})`

How can I move this into a module?  And Once there will Alpine.store still work on the html side?  

thanks in advance

**Comment by ekwoka** [comment]
By "into module" you mean into a separate file for bundling/module loaded and such?


It's recommended to use a kind of plugin function shaped like


```js
export const myStore = (Alpine) => Alpine.store(/** stuff */)
```


Then in your main script

```js
import Alpine from 'alpinejs'
import { myStore } from './myStore.js'

Alpine.plugin(myStore)

Alpine.start()
```

That's the best way to modularize all your components, stores, plugins, etc.

In your case, just make sure you do the persist plugin before the store.
